osu.Game/Screens/Edit/Compose/Components/Timeline/Timeline.cs

```diff
@@ -197,6 +189,22 @@ protected override void Update()
         {
             base.Update();
 
+            if (!trackWasLoaded && track.IsLoaded)
+            {
+                trackWasLoaded = true;
+
+                defaultTimelineZoom = getZoomLevelForVisibleMilliseconds(6000);
+                MaxZoom = getZoomLevelForVisibleMilliseconds(500);
+                MinZoom = getZoomLevelForVisibleMilliseconds(10000);
+                Zoom = (float)(defaultTimelineZoom * editorBeatmap.BeatmapInfo.TimelineZoom);
```

Comment:
Didn't notice this at first review, but it appears moving this logic from BDL to `Update()` causes `OnZoomChange` to be fired when changing the range, which in turn overwrite the beatmap's stored timeline zoom before it's passed to `Zoom`.